### DISCUSSION TITLE
OpenSSL QUIC Sample Client keeps polling for response from public QUIC servers (like youtube.com, cloudflare-quic.com, etc.)

### DISCUSSION THREAD

**Discussion by samitanwer** [openssl/openssl#24726] [Q&A] [answered]
I am trying to get the OSSL sample blocking and non-blocking clients for QUIC to work against public QUIC servers from MSFT, Google, FB, etc. The handshake seems to complete fine and I am able to send a GET Request over but while receiving data back from server the client seems to just keep polling for data in the `poll_two_fds()` method (quic_reactor.c).

I have went ahead and confirmed with the help of Wireshark that the GET request is sent to the server over the client-initiated bidirectional default stream (which is what the OSSL blocking and non-blocking clients are expected to do). As per the traces from Wireshark the servers just don't respond to the GET request. 

My GET request looks like -
"GET / HTTP/1.1\r\nUser-Agent: Mozilla/4.0 (compatible; MSIE5.01; Windows NT)\r\nHost: cloudflare-quic.com \r\nAccept-Language: en-us\r\nConnection: Close\r\n\r\n";

I have also tried to accept a new stream and listen on it for a response (using SSL_accept_stream()) without much luck. Any help would be greatly appreciated. 

The following is my wireshark trace [](url)
![image](https://github.com/openssl/openssl/assets/2293579/f2ed7328-ac76-4bc0-944a-693deba5bf52)

P.S I am able to get the client to receive response from the OSSL QUIC Sample server.

**Comment by samitanwer** [comment]
The GET request can be seen in the packet shown in the attached screenshot:
[
![image](https://github.com/openssl/openssl/assets/2293579/98288725-54e4-444f-9b5a-b5064d6a1d66)

**Comment by t8m** [comment, marked as answer]
The sample clients won't interoperate with public HTTP3 servers because the sample clients do not implement HTTP3. The only HTTP3 demo client we have is in demos/http3

**Reply by samitanwer** [reply]
Thanks @t8m for pointing me to the HTTP3 sample. The demo client appears to be specific to nghttp3, would it be right to assume that it won't be interoperable with other (server) implementations of HTTP3?

**Reply by t8m** [reply]
No, that is not right. nghttp3 should be interoperable with any http3 servers. nghttp3 is just a standalone implementation of http3 protocol that the demo client is using.

**Comment by mattcaswell** [comment]
> "GET / HTTP/1.1\r\nUser-Agent: Mozilla/4.0 (compatible; MSIE5.01; Windows NT)\r\nHost: cloudflare-quic.com \r\nAccept-Language: en-us\r\nConnection: Close\r\n\r\n";

This is an HTTP/1.1 request. Did you negotiate HTTP/1.1 in the ALPN? AFAIK there isn't a standard for HTTP/1.1 over QUIC??

**Reply by samitanwer** [reply]
I negotiated h3 in my ALPN but in my GET request I am using 'HTTP/1.1'. Like @t8m mentioned the sample does not fulfill HTTP3 requirements so this sample is probably not the right choice anyway so I will switch over to the HTTP3 demo client. Thanks @mattcaswell.
